### ISSUE TITLE
Build error after upgrading to RN 0.70.2

### ISSUE THREAD

**Issue by johnf** [software-mansion/react-native-reanimated#3644]
### Description

After upgrading to 0.70.2 I'm seeing the build error below

It looks like a path problem notice 
```/home/johnf/work/gladly/mobile/node_modules/react-native-reanimated/mobile/node_modules/react-native/package.json```
has an extra mobile in it after react-native-reanimated

```
> Configure project :react-native-firebase_messaging
:react-native-firebase_messaging package.json found at /home/johnf/work/gladly/mobile/node_modules/@react-native-firebase/messaging/package.json
:react-native-firebase_app package.json found at /home/johnf/work/gladly/mobile/node_modules/@react-native-firebase/app/package.json
:react-native-firebase_messaging:firebase.bom using default value: 30.5.0
:react-native-firebase_messaging package.json found at /home/johnf/work/gladly/mobile/node_modules/@react-native-firebase/messaging/package.json
:react-native-firebase_messaging:version set from package.json: 14.12.0 (14,12,0 - 14012000)
:react-native-firebase_messaging:android.compileSdk using custom value: 33
:react-native-firebase_messaging:android.targetSdk using custom value: 33
:react-native-firebase_messaging:android.minSdk using custom value: 23
:react-native-firebase_messaging:reactNativeAndroidDir /home/johnf/work/gladly/mobile/node_modules/react-native/android

Deprecated Gradle features were used in this build, making it incompatible with Gradle 8.0.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

See https://docs.gradle.org/7.5.1/userguide/command_line_interface.html#sec:command_line_warnings
5 actionable tasks: 5 executed

FAILURE: Build failed with an exception.

* Where:
Build file '/home/johnf/work/gladly/mobile/node_modules/react-native-reanimated/android/build.gradle' line: 135

* What went wrong:
A problem occurred evaluating project ':react-native-reanimated'.
> /home/johnf/work/gladly/mobile/node_modules/react-native-reanimated/mobile/node_modules/react-native/package.json (No such file or directory)

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 7s

error Failed to install the app. Make sure you have the Android development environment set up: https://reactnative.dev/docs/environment-setup.
Error: Command failed: ./gradlew app:installDebug -PreactNativeDevServerPort=8081 -PreactNativeDebugArchitectures=x86_64 -PreactNativeArchitectures=x86_64

FAILURE: Build failed with an exception.

* Where:
Build file '/home/johnf/work/gladly/mobile/node_modules/react-native-reanimated/android/build.gradle' line: 135

* What went wrong:
A problem occurred evaluating project ':react-native-reanimated'.
> /home/johnf/work/gladly/mobile/node_modules/react-native-reanimated/mobile/node_modules/react-native/package.json (No such file or directory)

```

### Steps to reproduce

1. Upgrade to RN 0.70.2
2. yarn android

### Snack or a link to a repository

N/A

### Reanimated version

3.0.0.rc2

### React Native version

0.70.2

### Platforms

Android

### JavaScript runtime

Hermes

### Workflow

React Native (without Expo)

### Architecture

Paper (Old Architecture)

### Build type

Debug mode

### Device

Android emulator

### Acknowledgements

Yes

**Comment by tomekzaw** [comment]
Hey, thanks for reporting this issue. Reanimated 2.10 doesn't work properly in monorepos on Android. Do you use one? If We really hope to get this fixed soon and released in 2.11.

**Comment by johnf** [comment]
@tomekzaw I'm using v3 and I'm not using a monorepo.

**Comment by Sub-Zero-1** [comment]
@johnf I had the same issue and it still persisted with the 2.11.0 and React-Native 0.73.0. The I read the hint of the error message (file: node_modules/react-native-reanimated/android/build.gradle): 

```
    throw new Exception(
        "[Reanimated] Unable to resolve react-native location in " +
        "node_modules. You should project extension property (in app/build.gradle) " +
        "`REACT_NATIVE_NODE_MODULES_DIR` with path to react-native."
    )
```
and applied my "specific" path to react-native in my "android/app/build.gradle" file as:

`def REACT_NATIVE_NODE_MODULES_DIR = "../../../node_modules/react-native"`

and now it works. Be aware, that my path could be different than your, as my project structure is different, so yours could be just: `def REACT_NATIVE_NODE_MODULES_DIR = "../../node_modules/react-native"`

Hope it helps.

**Comment by tomekzaw** [comment]
@Sub-Zero-1, what's your setup?

Instead of patching build.gradle, you can override the path by setting environmental variable `REACT_NATIVE_NODE_MODULES_DIR`.

**Comment by Sub-Zero-1** [comment]
@tomekzaw I did it. First it solved it, but later on it did not work anymore and retried it multiple times. I don't know why, but does not matter to my as the patch works, but now I have the multiple version conflict with 2.11.0 while pod install. I don't have the multiple version problem with the 2.10.0 so I will now try to adapt the 2.10.0 with the build gradle of the 2.11.0 including my patch as I need to upgrade to RN 0.70.3. pozdrawiam

**Comment by Sub-Zero-1** [comment]
Maybe this helps someone:

I tried to use 2.10.0, but I am using a monorepo, so I got the problem mentioned here from thread starter. Then I updated to 2.11.0 and I got the same error. I patched the version (with help of patch-package) with my specific path, but differently then mentioned in my thread before.

```
-    def reactNativeLocation = safeAppExtGet("REACT_NATIVE_NODE_MODULES_DIR", null)
+    def reactNativeLocation = safeAppExtGet("REACT_NATIVE_NODE_MODULES_DIR", "../../../../node_modules/react-native")
```

The option with setting REACT_NATIVE_NODE_MODULES_DIR environment variable mentioned by @tomekzaw did not work for me, or I did it wrong. It would be nice to get an actual example how to do that. I tried with: 

`def REACT_NATIVE_NODE_MODULES_DIR = "../../../../node_modules/react-native"` in build.gradle, but somehow it could not read it or it also searched in a wrong path for the build.gradle of my project?!. Maybe I did it wrong?

Then I could build Android and start the App. On iOS I had to do a `pod install`, but it **failed** with **multiple version conflict**, which I did not have with version 2.10.0. 

Therefore, I installed version 2.10.0 and replaced the node_modules/react-native-reanimated/android/build.gradle with the build.gradle from version 2.11.0 and of course adapted the path like mentioned above. Then I could run pod install and the App worked on Android and iOS.

**Comment by tomekzaw** [comment]
@Sub-Zero-1 `def REACT_NATIVE_NODE_MODULES_DIR = ...` won't work. You need to set environment variable and make sure it's actually used when building the app.

**Comment by shant1010** [comment]
Getting the same issue, also not using a monorepo

**Comment by johnf** [comment]
@tomekzaw I'm still seeing this in 3.0.0-rc3 for RN 0.70.4

From what I can tell this is due to the following
```
def isDeveloperMode() {
  // developer mode, to run Example app                                                                                                                                                                                                                                                                                                                                                                                             
  return file("$projectDir/../node_modules/react-native/package.json").exists()                                                                                                                                                                                                                                                                                                                                                     
}   

 def resolveReactNativeDirectory() {                        
   if (isDeveloperMode()) {                               
     return file("$projectDir/../${getPlaygroundAppName()}/node_modules/react-native")
   }                                                                                                   
   return file("$projectDir/../../react-native")
}
```

When ```isDeveloperMode``` returns ```true``` then ```resolveReactNativeDirectory``` injects the app name, in my case ```mobile``` into the path.

If I hard code ```isDeveloperMode``` to ```false```, everything builds fine.

Does ```isDeveloperMode``` exist just for the example app, or is it supposed to be true for dev builds?

**Comment by johnf** [comment]
This is fixed in [3.0.0-rc.4](https://github.com/software-mansion/react-native-reanimated/releases/tag/3.0.0-rc.4)
